src/shmem_team.c

```diff
@@ -371,33 +371,41 @@ int shmem_internal_team_split_2d(shmem_internal_team_t *parent_team, int xrange,
     const int num_xteams = ceil( parent_size / (float)xrange );
     const int num_yteams = xrange;
 
-    int start = parent_start;
+    int start = 0;
     int ret = 0;
 
+    shmem_internal_team_t *my_xteam, *my_yteam;
+
     for (int i = 0; i < num_xteams; i++) {
         int xsize = (i == num_xteams - 1 && parent_size % xrange) ? parent_size % xrange : xrange;
 
         ret = shmem_internal_team_split_strided(parent_team, start, parent_stride,
```

Comment:
Is the parent team's stride the correct thing to use here? I suspect we need some more unit tests to cover the various cases in here, for example where the parent team stride is more than 1.

Reply:
Yeah - I think it's correct.  I believe the "3D" split test (`spec-examples/shmem_team_split_2D.c`) covers this case during the 2nd split: In the 1st split, the parent stride (team_world) is 1, but in the 2nd split, the parent team has a stride of 3 in the x-splits, and a stride of 12 in the y-splits.

Reply:
Isn't the stride argument supposed to be relative to the PE numbers of the parent team? And isn't the parent team's stride relative to the global PE number space?

Reply:
I don't see any unit tests (in the branch of this PR, ignoring the awful spec test) that call `shmem_team_split_2d`?

Reply:
Yes, Yes, and... Yes?

This stride is relative to the parent team, which is relative to the global PE space...  maybe we should discuss in person, since I don't see an alternative stride that would make sense here.

Right, there are no unit tests with `split_2d` after moving the spec example, but I'm working on a 2D heat program, which might actually be a better fit for 'apps'?   Unfortunately, I don't believe a 2D heat example would exercise a non-unit stride in the parent, IIUC, but I believe the 3D split does.